proto/proto-backplane-grpc/src/main/proto/deephaven/proto/storage.proto

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+syntax = "proto3";
+
+package io.deephaven.proto.backplane.grpc;
+
+option java_multiple_files = true;
+option optimize_for = SPEED;
+option go_package = "github.com/deephaven/deephaven-core/go/internal/proto/storage";
+
+/*
+ * Shared storage management service.
+*/
+service StorageService {
+  rpc ListItems(ListItemsRequest) returns (ListItemsResponse) {}
+  rpc FetchFile(FetchFileRequest) returns (FetchFileResponse) {}
+  // Can create new files or modify existing
+  rpc SaveFile(SaveFileRequest) returns (SaveFileResponse) {}
+  rpc MoveItem(MoveItemRequest) returns (MoveItemResponse) {}
+  rpc CreateDirectory(CreateDirectoryRequest) returns (CreateDirectoryResponse) {}
+  rpc DeleteItem(DeleteItemRequest) returns (DeleteItemResponse) {}
+}
+
+message ListItemsRequest {
+  string path = 1;// the path to the directory to list. empty to list top level
+  optional string filter_glob = 2;// A pattern to filter for, with "*" as a wildcard
```

Comment:
More docs would be good here, this is currently implemented using https://docs.oracle.com/javase/8/docs/api/java/nio/file/FileSystem.html#getPathMatcher-java.lang.String- with the following restrictions:
 - no path separators allowed, so other directories can't be queried
 - no `**` so sub-directories can't be queried